src/Xiaomi_Scale.py

```diff
@@ -176,16 +149,21 @@ def _publish(self, weight, unit, mitdatetime, hasImpedance, miimpedance):
         message += ',"TimeStamp":"' + mitdatetime + '"'
         message += '}'
         try:
-          self.mqtt_client.publish(MQTT_PREFIX + '/' + user + '/weight', message, qos=1, retain=True)
-          sys.stdout.write('Sent data to topic %s: %s' % (MQTT_PREFIX + '/' + user + '/weight', message + '\n'))
+            sys.stdout.write('Sent data to topic %s: %s' % (MQTT_PREFIX + '/' + user + '/weight', message + '\n'))
+            publish.single(
+                MQTT_PREFIX + '/' + user + '/weight',
+                message,
+                qos=1,
+                retain=True,
+                hostname=MQTT_HOST,
+                port=MQTT_PORT,
+                auth={'username':MQTT_USERNAME, 'password':MQTT_PASSWORD}
+            )
         except:
-          sys.stdout.write('Could not publish to MQTT, Disconnecting...\n')
-          MQTT_CONNECTED = False
-          self.mqtt_client.disconnect()
-          pass
+          sys.stdout.write('Could not publish to MQTT\n')
+          raise
```

Comment:
why change pass to raise? I get that the primary function of this program is to send data to MQTT, but should we really stop it on first fail? A network issue could prevent a one off communication with the MQTT broker, which should be picked up on the next pass...

Reply:
We won't stop, exception will be catched in https://github.com/lolouk44/xiaomi_mi_scale/blob/4238d1af5c92b4cb2719362a29682b4b33eb62b9/src/Xiaomi_Scale.py#L212

This way we have 
- unable to publish
- an error happened during the script
(I'm on my phone, sorry for not copy paste real messages )

Deep down in the code, we should avoid except: pass if we can

Reply:
Thanks. Still got lots to learn indeed...